Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -673,6 +678,23 @@ theorem erase_zero (a : α) : erase a (0 : α →₀ M) = 0 := by
   classical rw [← support_eq_empty, support_erase, support_zero, erase_empty]
 #align finsupp.erase_zero Finsupp.erase_zero
 
+theorem erase_eq_update (f : α →₀ M) (a : α) : f.erase a = update f a 0 :=
+  letI := Classical.decEq α
+  ext fun _ => (Function.update_apply _ _ _ _).symm
+
+-- not `simp` as `erase_of_not_mem_support` can prove this
+theorem erase_idem (f : α →₀ M) (a : α) :
+    erase a (erase a f) = erase a f := by
+  rw [erase_eq_update, erase_eq_update, update_idem]
+
+@[simp] theorem update_erase (f : α →₀ M) (a : α) (b : M) :
+    update (erase a f) a b = update f a b := by
+  rw [erase_eq_update, update_idem]
+
+@[simp] theorem erase_update (f : α →₀ M) (a : α) (b : M) :
+    erase a (update f a b) = erase a f := by
```

Comment:
```suggestion
@[simp] theorem erase_update_self (f : α →₀ M) (a : α) (b : M) :
    erase a (update f a b) = erase a f := by
```

Reply:
On `Finset` this is called `erase_insert_eq_erase`